tmt/steps/report/reportportal.py

```diff
@@ -1,152 +1,221 @@
         default=os.environ.get("TMT_REPORT_REPORTPORTAL_TOKEN"),
-        help="The token to use for upload to the ReportPortal instance.")
-    project: Optional[str] = tmt.utils.field(
+        help="The token to use for upload to the ReportPortal instance (from RP user profile).")
+    project: str = field(
         option="--project",
-        metavar="PROJECT",
-        default=None,
+        metavar="PROJECT_NAME",
+        default="",
         help="The project name which is used to create the full URL.")
-    launch_name: Optional[str] = tmt.utils.field(
-        option="--launch-name",
-        metavar="NAME",
+    launch: Optional[str] = field(
+        option="--launch",
+        metavar="LAUNCH_NAME",
         default=None,
-        help="The launch name (base name of run id used by default).")
+        help="The launch name (name of plan per launch is used by default).")
+    tmt_vars: bool = field(
+        option="--tmt-vars",
+        default=False,
+        is_flag=True,
+        help="Report also the \"TMT_\" environment variables that are excluded by default.\
+(Recommended to have the tmt id generated to preserve the history aggregation for RP item.) ")
 
 
 @tmt.steps.provides_method("reportportal")
 class ReportReportPortal(tmt.steps.report.ReportPlugin):
     """
-    Report test results to a ReportPortal instance
+    Report test results to a ReportPortal instance via API.
 
     Requires a TOKEN for authentication, a URL of the ReportPortal
-    instance and the PROJECT name. In addition to command line options
-    it's possible to use environment variables to set the url and token:
+     instance and the PROJECT name. In addition to command line options
+     it's possible to use environment variables to set the url and token:
 
         export TMT_REPORT_REPORTPORTAL_URL=...
         export TMT_REPORT_REPORTPORTAL_TOKEN=...
 
-    The optional launch NAME is passed to ReportPortal. Assuming the URL
-    and TOKEN variables are provided by the environment, the config can
-    look like this:
+    The optional launch NAME doesn't have to be provided if it is the same
+     as the plan name (by default). Assuming the URL and TOKEN variables
+     are provided by the environment, the plan config can
+     look like this:
 
         report:
             how: reportportal
             project: baseosqe
-            launch-name: maven
+
+        context:
+            ...
+
+        environment:
+            ...
+
+    The context and environment sections must be filled in order to report
+    context as attributes and environment variables as parameters in the item details.
+    Other reported fmf data are summary, id, web link and contact per test.
     """
 
     _data_class = ReportReportPortalData
 
+    def handle_response(self, response: requests.Response) -> None:
+        if not response.ok:
+            raise tmt.utils.ReportError(
+                f"\nReceived non-ok status code from ReportPortal: {response.text}")
```

Comment:
Leading `\n` will mess with tmt's exception logging, as mentioned in https://github.com/teemtee/tmt/pull/2088#pullrequestreview-1465169856.